### PR TITLE
LPS-156767 Uses unique id map, since name map should be used as fallback

### PR DESCRIPTION
**Pull request by ealonso**

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by ealonso** [comment]
cc: @ruben-pulido

**Comment by ealonso** [comment]
ci:stop

**Comment by ealonso** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward

File: modules/apps/info/info-api/src/main/java/com/liferay/info/form/InfoForm.java [hunk 1]
```diff
@@ -231,10 +231,9 @@ private void _addToAllInfoFields(InfoFieldSetEntry infoFieldSetEntry) {
 
 			InfoFieldSet infoFieldSet = (InfoFieldSet)infoFieldSetEntry;
 
-			for (InfoFieldSetEntry childInfoFieldSetEntry :
-					infoFieldSet.getInfoFieldSetEntries()) {
-
-				_addToAllInfoFields(childInfoFieldSetEntry);
+			for (InfoField<?> infoField : infoFieldSet.getAllInfoFields()) {
+				_infoFieldsByName.put(infoField.getName(), infoField);
```
[the remark this answers is not in the document]
Comment (ealonso) [reply modules/apps/info/info-api/src/main/java/com/liferay/info/form/InfoForm.java:237]:
```java
public List<InfoField> getAllInfoFields() {
	List<InfoField> allInfoFields = new ArrayList<>();

	for (InfoFieldSetEntry infoFieldSetEntry :
			_builder._infoFieldSetEntries.values()) {

		if (infoFieldSetEntry instanceof InfoField) {
			allInfoFields.add((InfoField)infoFieldSetEntry);
		}
		else if (infoFieldSetEntry instanceof InfoFieldSet) {
			InfoFieldSet infoFieldSet = (InfoFieldSet)infoFieldSetEntry;

			allInfoFields.addAll(infoFieldSet.getAllInfoFields());
		}
	}

	return allInfoFields;
}
```

[the remark this answers is not in the document]
Comment (ruben-pulido) [reply modules/apps/info/info-api/src/main/java/com/liferay/info/form/InfoForm.java:237]: You're right! Then we're good

**Comment by ealonso** [comment]
ci:reevaluate:1307401_3638

**Comment by ealonso** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:reevaluate:1317401_3901

**Comment by ealonso** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward

**Comment by ruben-pulido** [comment]
ci:reevaluate:1277401_3810

**Comment by ruben-pulido** [comment]
ci:forward

**Pull request closed without merging**